pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1040,19 +1027,30 @@ func setNodeSelectors(tx *gorm.DB, req *datastore.SetNodeSelectorsRequest) (*dat
 	return &datastore.SetNodeSelectorsResponse{}, nil
 }
 
-func getNodeSelectors(tx *gorm.DB, req *datastore.GetNodeSelectorsRequest) (*datastore.GetNodeSelectorsResponse, error) {
-	var models []NodeSelector
-	if err := tx.Where("spiffe_id = ?", req.SpiffeId).Find(&models).Error; err != nil {
+func getNodeSelectors(ctx context.Context, dbType string, db *sql.DB, req *datastore.GetNodeSelectorsRequest) (*datastore.GetNodeSelectorsResponse, error) {
+	query := "SELECT type, value FROM node_resolver_map_entries WHERE spiffe_id=? ORDER BY id"
+	if dbType == PostgreSQL {
+		query = postgreSQLRebind(query)
+	}
+	rows, err := db.QueryContext(ctx, query, req.SpiffeId)
+	if err != nil {
 		return nil, sqlError.Wrap(err)
 	}
+	defer rows.Close()
 
 	var selectors []*common.Selector
-	for _, selector := range models {
-		selectors = append(selectors, &common.Selector{
-			Type:  selector.Type,
-			Value: selector.Value,
-		})
+	for rows.Next() {
+		selector := new(common.Selector)
```

Comment:
nit: our internal style guide recommends `&common.Selector{}` over `new(common.Selector)` as it makes it look similar if there are fields or not. Feel free to ignore.